proxy/src/coco/source.rs

```diff
@@ -260,18 +281,66 @@ pub fn local_state(repo_path: &str) -> Result<LocalState, error::Error> {
     Ok(LocalState { branches, managed })
 }
 
-/// Retrieves the [`Commit`] for the given `sha1`.
+/// Retrieves the [`CommitHeader`] for the given `sha1`.
 ///
 /// # Errors
 ///
 /// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit, error::Error> {
+pub fn commit_header<'repo>(
+    browser: &mut Browser<'repo>,
+    sha1: &str,
+) -> Result<CommitHeader, error::Error> {
     browser.commit(surf::vcs::git::Oid::from_str(sha1)?)?;
 
     let history = browser.get();
     let commit = history.first();
 
-    Ok(Commit::from(commit))
+    Ok(CommitHeader::from(commit))
+}
+
+/// Retrieves a [`Commit`].
+///
+/// # Errors
+///
+/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
+pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit, error::Error> {
+    let oid = surf::vcs::git::Oid::from_str(sha1)?;
+    browser.commit(oid)?;
+
+    let history = browser.get();
+    let commit = history.first();
+
+    let diff = if let Some(parent) = commit.parents.first() {
+        browser.diff(*parent, oid)?
+    } else {
+        browser.initial_diff(oid)?
+    };
+
+    let mut deletions = 0;
+    let mut additions = 0;
+
+    for file in diff.modified.iter() {
```

Comment:
This could probably be expressed with some `Iterator` combinators, but not sure if it's worth it :thinking:

Reply:
Yeah I know, but don't think it's worth it heh.